3-UsingAPIs/Week1/homework/ex5-pokerDiceChain.js

```diff
@@ -16,13 +16,18 @@ const rollDie = require('../../helpers/pokerDiceRoller');
 
 function rollDice() {
   const results = [];
+  const returnResults = (num) => {
+    return (value) => {
+      results.push(value);
+      return rollDie(num);
+    };
+  };
```

Comment:
Ah, a higher-order version. This is the second time I see this solution in the homework this week. Makes the code where it is called short and sweet. You can even shorten this to:

```suggestion
  const returnResults = (num) => (value) => {
      results.push(value);
      return rollDie(num);
  };
```